libs/remix-core-plugin/src/lib/compiler-artefacts.ts

```diff
@@ -1,105 +1,83 @@
 'use strict'
-import { Plugin } from '@remixproject/engine'
-import { util } from '@remix-project/remix-lib'
-import { CompilerAbstract } from '@remix-project/remix-solidity'
+import {Plugin} from '@remixproject/engine'
+import {util} from '@remix-project/remix-lib'
+import {CompilerAbstract} from '@remix-project/remix-solidity'
 
 const profile = {
   name: 'compilerArtefacts',
-  methods: ['get', 'addResolvedContract', 'getCompilerAbstract', 'getAllContractDatas', 'getLastCompilationResult', 'getArtefactsByContractName', 'getContractDataFromAddress', 'getContractDataFromByteCode', 'saveCompilerAbstract'],
+  methods: ['get', 'addResolvedContract', 'getCompilerAbstract', 'getAllContractDatas', 'getLastCompilationResult', 'getArtefactsByContractName', 'getContractDataFromAddress', 'getContractDataFromByteCode', 'saveCompilerAbstract', 'getAllCompilerAbstracts'],
   events: [],
-  version: '0.0.1'
+  version: '0.0.1',
 }
 
 export class CompilerArtefacts extends Plugin {
   compilersArtefactsPerFile: any
   compilersArtefacts: any
-  constructor () {
+  constructor() {
     super(profile)
     this.compilersArtefacts = {}
     this.compilersArtefactsPerFile = {}
   }
 
-  clear () {
+  clear() {
     this.compilersArtefacts = {}
     this.compilersArtefactsPerFile = {}
   }
 
-  saveCompilerAbstract (file: string, compilerAbstract: CompilerAbstract) {
+  saveCompilerAbstract(file: string, compilerAbstract: CompilerAbstract) {
     this.compilersArtefactsPerFile[file] = compilerAbstract
   }
 
-  onActivation () {
-    const saveCompilationPerFileResult = (file, source, languageVersion, data, input?) => {
+  getAllCompilerAbstracts() {
+    return this.compilersArtefactsPerFile
+  }
+
+  onActivation() {
+    const saveCompilationResult = (file, source, languageVersion, data, input?) => {
       this.compilersArtefactsPerFile[file] = new CompilerAbstract(languageVersion, data, source, input)
+      this.compilersArtefacts.__last = this.compilersArtefactsPerFile[file]
```

Comment:
Refactored the contents of listener functions to this single function called `saveCompilationResult`. This also contains what the prev. `saveCompilationPerFileResult` has.

Basically we always pass the `input` to the `CompilerAbstract` constructor and then point this one as the `__last` artefact. Maybe we might want to create a separate new CompilerAbstract for `__last` but this seemed proper to me.e